subsystem/cluster_test.go

```diff
@@ -176,41 +177,63 @@ var _ = Describe("Cluster tests", func() {
+	Expect(err).NotTo(HaveOccurred())
+	c := rep.GetPayload()
+	if swag.StringValue(c.Status) == state {
+		return stateInfo == IgnoreStateInfo || swag.StringValue(c.StatusInfo) == stateInfo
+	}
+
+	return false
+}
+
 func waitForClusterState(ctx context.Context, clusterID strfmt.UUID, state string, timeout time.Duration, stateInfo string) {
 	log.Infof("Waiting for cluster %s status %s", clusterID, state)
-	for start := time.Now(); time.Since(start) < timeout; {
-		rep, err := userBMClient.Installer.GetCluster(ctx, &installer.GetClusterParams{ClusterID: clusterID})
-		Expect(err).NotTo(HaveOccurred())
-		c := rep.GetPayload()
-		if swag.StringValue(c.Status) == state {
-			break
+
+	for start, successInRow := time.Now(), 0; time.Since(start) < timeout; {
+		if isClusterInState(ctx, clusterID, state, stateInfo) {
+			successInRow++
+		} else {
+			successInRow = 0
 		}
+
+		// Wait for cluster state to be consistent
+		if successInRow >= minSuccessesInRow {
+			return
+		}
+
 		time.Sleep(time.Second)
 	}
-	rep, err := userBMClient.Installer.GetCluster(ctx, &installer.GetClusterParams{ClusterID: clusterID})
+
+	Fail(fmt.Sprintf("Cluster %s wasn't in state %s for %d times in a row", clusterID, state, minSuccessesInRow))
+}
+
+func isHostInState(ctx context.Context, clusterID strfmt.UUID, hostID strfmt.UUID, state string) bool {
+	rep, err := userBMClient.Installer.GetHost(ctx, &installer.GetHostParams{ClusterID: clusterID, HostID: hostID})
 	Expect(err).NotTo(HaveOccurred())
 	c := rep.GetPayload()
-	Expect(swag.StringValue(c.Status)).Should(Equal(state))
-	if stateInfo != IgnoreStateInfo {
-		Expect(swag.StringValue(c.StatusInfo)).Should(Equal(stateInfo))
-	}
+	return swag.StringValue(c.Status) == state
 }
 
 func waitForHostState(ctx context.Context, clusterID strfmt.UUID, hostID strfmt.UUID, state string, timeout time.Duration) {
 	log.Infof("Waiting for host %s state %s", hostID, state)
-	for start := time.Now(); time.Since(start) < timeout; {
-		rep, err := userBMClient.Installer.GetHost(ctx, &installer.GetHostParams{ClusterID: clusterID, HostID: hostID})
-		Expect(err).NotTo(HaveOccurred())
-		c := rep.GetPayload()
-		if swag.StringValue(c.Status) == state {
-			break
+	for start, successInRow := time.Now(), 0; time.Since(start) < timeout; {
+		if isHostInState(ctx, clusterID, hostID, state) {
+			successInRow++
+		} else {
+			successInRow = 0
+		}
+
+		// Wait for host state to be consistent
+		if successInRow >= minSuccessesInRow {
+			return
 		}
+
 		time.Sleep(time.Second)
 	}
-	rep, err := userBMClient.Installer.GetHost(ctx, &installer.GetHostParams{ClusterID: clusterID, HostID: hostID})
-	Expect(err).NotTo(HaveOccurred())
-	c := rep.GetPayload()
-	ExpectWithOffset(1, swag.StringValue(c.Status)).Should(Equal(state))
+
+	Fail(fmt.Sprintf("Host %s in Cluster %s wasn't in state %s for %d times in a row", hostID, clusterID, state, minSuccessesInRow))
```

Comment:
it will be nice to see the current state, so it will be clear to understand what is expected
Otherwise it will be hard to understand the flow:
```
Cluster 277b2c24-3ef0-4e44-8fb8-7bd91a1323a0 wasn't in state insufficient for 2 times in a row
```

Reply:
Added